website/docs/r/wafv2_web_acl.html.markdown

```diff
@@ -0,0 +1,495 @@
+
+The `field_to_match` block supports the following arguments:
+
+* `all_query_arguments` - (Optional) Inspect all query arguments.
+* `body` - (Optional) Inspect the request body, which immediately follows the request headers.
+* `method` - (Optional) Inspect the HTTP method. The method indicates the type of operation that the request is asking the origin to perform.
+* `query_string` - (Optional) Inspect the query string. This is the part of a URL that appears after a `?` character, if any.
+* `single_header` - (Optional) Inspect a single header. See [Single Header](#single-header) below for details.
+* `single_query_argument` - (Optional) Inspect a single query argument. See [Single Query Argument](#single-query-argument) below for details.
+* `uri_path` - (Optional) Inspect the request URI path. This is the part of a web request that identifies a resource, for example, `/images/daily-ad.jpg`.
+
+### Single Header
+
+Inspect a single header. Provide the name of the header to inspect, for example, `User-Agent` or `Referer` (provided as lowercase strings).
+
+The `single_header` block supports the following arguments:
+
+* `name` - (Optional) The name of the query header to inspect. This setting must be provided as lower case characters.
+
+### Single Query Argument
+
+Inspect a single query argument. Provide the name of the query argument to inspect, such as `UserName` or `SalesRegion` (provided as lowercase strings).
+
+The `single_query_argument` block supports the following arguments:
+
+* `name` - (Optional) The name of the query header to inspect. This setting must be provided as lower case characters.
+
+### Text Transformation
+
+The `text_transformation` block supports the following arguments:
+
+* `priority` - (Required) The relative processing order for multiple transformations that are defined for a rule statement. AWS WAF processes all transformations, from lowest priority to highest, before inspecting the transformed content.
+* `type` - (Required) The transformation to apply, you can specify the following types: `NONE`, `COMPRESS_WHITE_SPACE`, `HTML_ENTITY_DECODE`, `LOWERCASE`, `CMD_LINE`, `URL_DECODE`. See the [documentation](https://docs.aws.amazon.com/waf/latest/APIReference/API_TextTransformation.html) for more details.
+
+### Visibility Configuration
+
+The `visibility_config` block supports the following arguments:
+
+* `cloudwatch_metrics_enabled` - (Required) A boolean indicating whether the associated resource sends metrics to CloudWatch. For the list of available metrics, see [AWS WAF Metrics](https://docs.aws.amazon.com/waf/latest/developerguide/monitoring-cloudwatch.html#waf-metrics).
+* `metric_name` - (Required) A friendly name of the CloudWatch metric. The name can contain only alphanumeric characters (A-Z, a-z, 0-9) hyphen(-) and underscore (_), with length from one to 128 characters. It can't contain whitespace or metric names reserved for AWS WAF, for example `All` and `Default_Action`.
```

Comment:
Some Markdown parsers don't special-case a single underscore character, so italicize the remainder of the document. This can be escaped by using `\_` instead